[16.0][MIG] controller_report_xls: Migration to 16.0. t#87779

CDanielSanSeg opened the pull request:
[MIG] controller_report_xls: Migration to 16.0

hbto committed:
[IMP] Controller module to export QWeb reports into XLS format has landed

hbto committed:
[PYLINT]

Commit:
[FIX] When we make a super we need to send the original values of the funcion, to avoid mistake for minimal changes in their values

hbto committed:
[IMP] Giving proper filename to report in XLS format

hbto committed:
[PYLINT]

Commit:
[FIX] Added new variables to work with the differents way to generate the url to create the reports

nhomar committed:
Add # -*- coding: utf-8 -*- may be it is necesary for pylint

Trying to fix this pylint errors.

cmd: [u'/home/runbot/instance/extra_addons/odoo-extra/runbot/static/build/35660-8-0-8-0-lo/pylint_run.sh']
[7;33m************* Module openerp.addons.controller_report_xls.controllers.main[0m
openerp.addons.controller_report_xls.controllers.main:2: [F0401([1;4;31mimport-error[0m), ] [1;4;31mUnable to import 'openerp.addons.web.http'[0m
openerp.addons.controller_report_xls.controllers.main:3: [E0611([1;31mno-name-in-module[0m), ] [1;31mNo name 'url_decode' in module 'werkzeug'[0m

moylop260 committed:
[REF] controller_report_xls: Disable pylint error. More info here: https://github.com/Vauxoo/addons-vauxoo/pull/102#issuecomment-73302691

hbto committed:
[IMP] <th/> tag sometimes is used as <td/> tag in some reports and they
need to be translated into excel as a cell

zaoral committed:
[ADD] Move controller_report_xls module description from the openerp descriptor to a README.md file.

zaoral committed:
[IMP] Add controller_report_xls module index.html file taking into account the README.md
description an a predefined module descriptor footer (Footer includes an
image with Vauxoo Large Logo.

This was created using an script Vauxoo/gist-vauxoo/description2html.

zaoral committed:
[IMP] Add missing controller_report_xls module icon. This icon is the Vauxoo Logo.

This was apply using the script Vauxoo/gist-vauxoo/description2html.

nhomar committed:
[IMP] changing versions forcing *.5 on versions to have them as reference.
[IMP] .version.py now works with ReGex

zaoral committed:
[IMP] Move controller_report_xls module description from README.md to README.rst file

nhomar committed:
[FIX] external_dependency problem.

zaoral committed:
[IMP] Update controller_report_xls module index.html file. New footer which image is reference from aws.

zaoral committed:
[IMP] Delete not used footer image in controller_report_xls module description at index.html file

zaoral committed:
[IMP] Update controller_report_xls module version +0.1 points to ensure that odoo apps refresh the new index.html description

luistorresm committed:
[IMP][controller_report_xls]Change BeautifulSoup by lxml.html

luistorresm committed:
[IMP][controller_report_xls]Fixed pylint

luistorresm committed:
[IMP][controller_report_xls]Reverted use to lxml by BeautifulSoup

moylop260 committed:
[REF] all py: Delete vim comment, add coding comment with pep8 style

moylop260 committed:
[REF] all py: Remove magic comment interpreter

moylop260 committed:
'[REF] addons-vauxoo: Standardize project with odoo guidelines.'

luistorresm committed:
[IMP][controller_report_xls]Changed bs4 by lxml

Commit:
[FIX] Wrong manifest version format

Commit:
[FIX] oca-autopep8: Fix bad-docstring-quotes and docstring-first-line-empty

KarenKawaii committed:
[ADD] controller_report_xls: Add missing translation files. (#834)

moylop260 committed:
[MERGE] Improve controller report xls Vauxoo/addons-vauxoo#860 by @feketemihai
 - [REF] aging_due_report: Migration to new api, small fixes
 - [IMP] controller_report_xls: take in consideration thead, tbody, map some the styles from css to xfstyle

hbto committed:
[IMP] Adding Number/Text Formatting

Commit:
[REF] deactivate non-migrated to 10.0 modules

Commit:
[MIG] controller_report_xls: Migration to 11.0

carralc committed:
[IMP] controller_report_xls: black, isort, prettier

carralc committed:
[MIG] controller_report_xls: Migration to 15.0

CarmenMiranda committed:
[FIX] controller_report_xls: Get Odoo Style

When the node had a style like
"background-color: #99CCFF; font-weight: bold;" the style that comes
after the "; " (with the white-space after the ;) was not being found on
the method `css2excel`.
That happened because the `get_odoo_style` returned the styles like:
`{'background-color': '#99CCFF', ' font-weight': 'bold'}` and the
`process_css` didn't find ' font-weight' in its keys.

So now cleaning the values on the `get_odoo_style` by removing the
whitespaces from the beginning and the end of the string, the method
will now return `{'background-color': '#99CCFF', 'font-weight': 'bold'}`
and the `process_css` will find 'font-weight' in its keys.

CarmenMiranda committed:
[FIX] controller_report_xls: Font Height

Before this change there was a traceback when a `font-size` was added:
```python
File "/usr/local/lib/python3.8/dist-packages/xlwt/BIFFRecords.py", line 725, in __init__
    self._rec_data = pack('<5H4B%ds' % uname_len, height, options, colour_index, weight, escapement,
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/odoo/instance/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/instance/odoo/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
struct.error: required argument is not an integer
```

That happened because the `height` expected for the fonts is an integer
value and the `get_font_height` was returning `float` values, for
example: "125.0", now we are rounding the new size to 0 decimals to
return the closest integer for that size.

hugho-ad reviewed (approved):
LGTM :+1:

luisg123v commented:
Please fix PR title.

hugho-ad commented:
@luisg123v 
done

luisg123v commented on controller_report_xls/controllers/main.py:
```diff
@@ -30,8 +30,8 @@
-from openerp.addons.report.controllers import main
-from openerp.addons.web.http import request, route
+from odoo.addons.report.controllers import main
+from odoo.addons.web.http import request, route
```
This is an "issue" coming from the previous version, it therefore could be fixed there. This is not strictly part of the migration.

luisg123v commented on controller_report_xls/controllers/main.py:
```diff
@@ -30,9 +30,10 @@
+
 from odoo.addons.report.controllers import main
 from odoo.addons.web.http import request, route
-from werkzeug import url_decode
```
This is autoformatting, it should be in a dedicated commit where `pre-commit-vauxoo` autofixes are applied. Besides, this could be applied in the previous version.

Same for other autoformatting changes.

luisg123v commented on controller_report_xls/controllers/main.py:
```diff
@@ -30,9 +30,10 @@
 import xlwt
 from cssutils import parseString
 from lxml import etree
+from werkzeug import url_decode
```
Commit title: module name is incomplete

luisg123v commented on controller_report_xls/controllers/main.py:
```diff
@@ -43,6 +44,7 @@ def get_css_style(csstext, style):
     cssutils.log.setLevel(logging.CRITICAL)
     cssnode = parseString(csstext)
     stylesheet = cssnode.cssRules
```
Move to the previous version.

luisg123v commented on controller_report_xls/requirements.txt:
```diff
@@ -0,0 +1,2 @@
+# To avoid warnings in the module controller_report_xls
+cssutils==2.7.1
```
- Does this file do anything? Requirements file should be placed at the top of the repository.
- Use `test-requirements.txt` instead, so the check to avoid requirements file doesn't complain
- Adjust title to use `ADD` and to mention the requirements file as module

luisg123v replied:
Missing the following part:

> mention the requirements file as module

I mean, the touched module in this case is the requirements file itself, not the module.

BTW, in both the requirements and the commit message you say it's to avoid warnings. I think that is not accurate because, AFAIK, if the requirement is not meet, the module fails to install. In other words, it's not a warning but an actual error.

I'd say instead:

```
[ADD] test-requirements.txt: add cssutils as requirement

Because it's required by the module `controller_report_xls`.
```

And in the requirements:

`# Required by controller_report_xls`

luisg123v commented on controller_report_xls/i18n/es_ES.po:
```diff
@@ -0,0 +1,15 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 8.0\n"
```
The file `es.po` should be regenerated as part of the migration. Besides, remove file `es_ES.po` and similar ones in the previous version.

luisg123v replied:
File was not regenerated in this version, still pointing to 15.0.

luisg123v commented on controller_report_xls/controllers/main.py:
```diff
@@ -0,0 +1,265 @@
+###########################################################################
+#    Module Writen to OpenERP, Open Source Management Solution
```
Remove all headers, in the previous version. This is a very old syntax.

luisg123v commented on controller_report_xls/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "version": "16.0.1.0.0",
+    "author": "Vauxoo",
+    "category": "Tools",
+    "website": "http://www.vauxoo.com/",
```
https

luisg123v commented on controller_report_xls/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "author": "Vauxoo",
+    "category": "Tools",
+    "website": "http://www.vauxoo.com/",
+    "license": "AGPL-3",
```
LGPL-3

luisg123v commented on controller_report_xls/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "css": [],
+    "qweb": [],
+    "installable": True,
+    "auto_install": False,
```
Remove all empty keys, see:

https://git.vauxoo.com/vauxoo/project-template/-/merge_requests/129

luisg123v commented:
Missing task ID on PR title

CDanielSanSeg committed:
[IMP] controller_report_xls: general improvements and and code cleanup

- Updated __manifest__.py: set license to LGPL-3, switched to HTTPS for
  URL, removed unnecessary keys.
- Removed outdated copyright header from main controller.
- Refreshed es.po with current project metadata.
- Deleted redundant translation files (es_ES, es_MX, es_PA, es_VE) to
  avoid duplication.

CDanielSanSeg committed:
[IMP] controller_report_xls: Apply pre-commit-vauxoo fixes

- Fix undefined variable 'cssstyle' in get_css_style function
- Fix undefined variable 'result' in match_color_index function
- Improve string comparison using truthiness instead of empty string
  check

CDanielSanSeg committed:
[REM] controller_report_xls: remove legacy controller

- Remove controllers/main.py, as the controller logic has been
  consolidated into reports/report_xls.py.
- Update imports to reflect the new structure.
- Modernize the codebase while maintaining backward compatibility for
  XLS report generation.

CDanielSanSeg committed:
[MIG] controller_report_xls: Migration to 16.0

CDanielSanSeg committed:
[ADD] test-requirements.txt: add cssutils as requirement

Because it's required by the module `controller_report_xls`.

CDanielSanSeg commented:
@luisg123v can you check this MR?

hugho-ad commented:
@luisg123v 

comments Done

luisg123v reviewed (approved):
LGTM :+1:
